components/src/store/reducers/banners.ts

```diff
@@ -0,0 +1,21 @@
+import { BannersAction, TypeKeys } from "../actions/index";
```

Comment:
It looks like with this approach we could end up with an infinite amount of names stored if I am understanding correctly. I think we will want to avoid something like this

Reply:
Correct, but this is based on the practical and hopefully reasonable assumption that we aren't going to be pummeling users with constantly running, constantly changing banners for all eternity. 😄 If it's really a concern, though, we can just date them and clean them up periodically. I can add the code to do that now or come back to it later. To me it'd be fine to do that later.